### PR TITLE
[CU-33zbdbb] - use total_issuance

### PR DESCRIPTION
**Pull request by andresmechali**
## Issue
https://app.clickup.com/t/33zbdbb

### CODE REVIEW

**Comment by ComposableFiB0t** [comment]
Task linked: [CU-33zbdbb Use `Inspect::total_issuance` instead of `RewardPool.total_shares`](https://app.clickup.com/t/33zbdbb)

File: code/parachain/frame/staking-rewards/src/lib.rs [hunk 1]
```diff
@@ -1316,6 +1314,8 @@ pub mod pallet {
 
 				if let Some(inflation) = stake.reductions.get_mut(reward_asset_id) {
 					*inflation += claim;
+				} else {
+					defensive!();
```
Comment (benluelo) [inline code/parachain/frame/staking-rewards/src/lib.rs:1318]: should add a message in the macro or a comment explaining why this is here

File: code/parachain/frame/staking-rewards/src/test/mod.rs [hunk 2]
```diff
@@ -796,6 +797,7 @@ fn unstake_in_case_of_not_zero_claims_and_early_unlock_should_work() {
 		process_and_progress_blocks::<StakingRewards, Test>(1);
 
 		mint_assets([BOB], [PICA::ID], PICA::units(200));
+		mint_assets([BOB], [XPICA::ID], XPICA::units(200));
```
Comment (benluelo) [inline code/parachain/frame/staking-rewards/src/test/mod.rs:800]: you should not be minting any xtokens in the tests, xtokens are only to be minted by the pallet

**Review by benluelo — CHANGES_REQUESTED** [review]

**Review by PoisonPhang — APPROVED** [review]
LGTM
However, we should remember that this partially depends on #1857 to ensure that the Share Asset is only controlled by pallet staking rewards.

File: code/parachain/frame/staking-rewards/src/lib.rs [hunk 3]
```diff
@@ -1197,7 +1194,8 @@ pub mod pallet {
 			fnft_account: &AccountIdOf<T>,
 		) -> DispatchResult {
 			T::Assets::mint_into(share_asset_id, fnft_account, awarded_shares)?;
-			T::Assets::set_lock(T::LockId::get(), share_asset_id, fnft_account, awarded_shares)
+			T::Assets::set_lock(T::LockId::get(), share_asset_id, fnft_account, awarded_shares)?;
```
Comment (vimukthi-git) [inline code/parachain/frame/staking-rewards/src/lib.rs:1197]: :+1:

**Review by vimukthi-git — APPROVED** [review]
LGTM

**Comment by andresmechali** [comment]
@mergify refresh

**Pull request merged**